### PR TITLE
ORCHESTRATIONS: Mesh Orchestration - Retrieve Messages

### PR DESCRIPTION
**Pull request by davidhayes03**

### CODE REVIEW

File: NEL.MESH.Tests.Unit/Services/Orchestrations/Mesh/MeshOrchestrationTests.cs [hunk 1]
```diff
@@ -46,6 +46,17 @@ private static string GetRandomString(int wordCount = 0)
         private static int GetRandomNumber() =>
             new IntRange(min: 2, max: 10).GetValue();
 
+        private static List<string> GetRandomMessages(int count)
```
Comment (cjdutoit) [inline NEL.MESH.Tests.Unit/Services/Orchestrations/Mesh/MeshOrchestrationTests.cs:49]:
```suggestion
 private static List<string> GetRandomMessages()
 {
    return Enumerable.Range(1, GetRandomNumber()).Select(item => GetRandomString())
 }
```

File: NEL.MESH.Tests.Unit/Services/Orchestrations/Mesh/MeshOrchestrationTests.Logic.RetrieveMessages.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+{
+    public partial class MeshOrchestrationTests
+    {
+        [Fact]
+        public async Task ShouldDoRetrieveMessagesAsync()
+        {
+            // given
+            string randomToken = GetRandomString();
+
+            List<string> outputMessages = GetRandomMessages(GetRandomNumber());
```
Comment (cjdutoit) [inline NEL.MESH.Tests.Unit/Services/Orchestrations/Mesh/MeshOrchestrationTests.Logic.RetrieveMessages.cs:21]:
```suggestion            
    List<string> outputMessages = GetRandomMessages();
```

File: NEL.MESH/Services/Orchestrations/Mesh/MeshOrchestrationService.cs [hunk 3]
```diff
@@ -72,7 +72,13 @@ public MeshOrchestrationService(ITokenService tokenService, IMeshService meshSer
             });
 
         public ValueTask<List<string>> RetrieveMessagesAsync() =>
-            throw new System.NotImplementedException();
+            TryCatch(async () =>
+            {
+                string token = await this.tokenService.GenerateTokenAsync();
+                List<string> outputMessage = await this.meshService.RetrieveMessagesAsync(authorizationToken: token);
```
Comment (cjdutoit) [inline NEL.MESH/Services/Orchestrations/Mesh/MeshOrchestrationService.cs:78]: You are missing the validation.  Add `ValidateToken(token);` like above and add a unit test for it

**Review by cjdutoit — CHANGES_REQUESTED** [review]
Missing validation

File: NEL.MESH/Services/Orchestrations/Mesh/MeshOrchestrationService.cs (hunk 3 above)
Comment (davidhayes03) [reply NEL.MESH/Services/Orchestrations/Mesh/MeshOrchestrationService.cs:78]: good spot

**Review by cjdutoit — CHANGES_REQUESTED** [review]
Still need to do the validation test for token

**Review by cjdutoit — APPROVED** [review]

**Pull request merged**